libraries/chain/transaction_context.cpp

```diff
@@ -431,14 +430,17 @@ namespace eosio { namespace chain {
       if( explicit_billed_cpu_time || pseudo_start != fc::time_point() ) return; // either irrelevant or already running
 
       auto now = fc::time_point::now();
+      auto paused = now - paused_time;
+      total_paused_time += paused;
+
       pseudo_start = now - billed_time;
-      if( (pseudo_start + billing_timer_duration_limit) <= deadline ) {
-         _deadline = pseudo_start + billing_timer_duration_limit;
-         deadline_exception_code = billing_timer_exception_code;
-      } else {
-         _deadline = deadline;
-         deadline_exception_code = deadline_exception::code_value;
+      _deadline += total_paused_time;
```

Comment:
No. The calls are guarded by `psuedo_start` being assigned `fc::time_point()` and `now - billed_time`.

Reply:
What I mean is, if a single transaction needs to compile more than one wasm, it look something like:
pause_resume 1:
```
total_paused_time = compile1
_deadline = original_deadline + compile1
```
pause/resume 2:
```
total_paused_time = compile1 + compile2
_deadline = original_deadline + compile1 + compile1 + compile2
```

because `total_paused _time` is already cumulative and we're accumulating again in `_deadline`.

Reply:
ah, yes. Thanks!